docs/extenders/Error Handling.md

```diff
@@ -0,0 +1,38 @@
+# Error Handling Best Practices v1
+
+This document describes how extenders using the Zowe Explorer API should report errors which occur in their extensions.
+
+TODO: The recommendations from this document will be solidified into an API in V2.
+
+## Contents
+
+- [Objectives](#objectives)
+- [Error Message Format](#error-message-format)
+- [Logging of Errors](#logging-of-errors)
+
+## Objectives
+
+The practices described here are intended to do the following:
+
+- Standardize the number & appearance of errors across Zowe Explorer and extenders using its API.
+- Help users understand the source of errors they encounter, by including the name of the extender in the error message.
+- Increase information available to developers in error logs.
+
+## Error Message Format
+
+Any error message should have the following format:
+
+```
+Extension Name Error - Error Message
+```
+
+Ex. for the Zowe Explorer FTP extension, the message would look like:
+
+```
+Zowe Explorer FTP Extension Error - Unable to delete node…
+```
+
+## Logging of Errors
+
+- Every error should be logged to the VSCode console. _TODO: Extenders need access to the Imperative logger!_
```

Comment:
When you say VSCode console, did you mean something like this?

```
globals.LOG.error(
            localize("initialize.log.error", "Error encountered while activating and initializing logger! ") +
                JSON.stringify(err)
        );
```

Reply:
I suggest to stick to what ZE is doing. We are logging all our error in the .vscode/logs folder. If this is the case, how will extenders do this?

Reply:
@jellypuno this is why it says, "TODO: Extenders need access to the Imperative logger!"

I am not sure what you want to do about this. I think adding access to the Imperative logger for extenders, via the ZE API, should go in another PR.

Reply:
I can add the code you posted as an example, but since this isn't what extenders will use (they would use some API function that is yet-to-be-created) I think there is not much point in doing so now. We will have to change this file when the error logging function is added to the API anyhow.

Reply:
For me, I am missing the answer to these questions:

- What is this for? 
- Why is this important?
- How am I going to accomplish this?

All I see are the TODOs and what are our future plans. What are we currently using today? Can extenders still have error handling best practices even if our future plans are not implemented yet? If yes, how are they going to do that?

Reply:
@jellypuno I adjusted the document, I think it is a bit more readable now. And the TODOs have been removed, hopefully they don't get forgotten!

Please review again if you have time ^^

Reply:
@katelynienaber Thanks for the updates! Do you want to create an issue for the TODOs? (Or maybe add them to issue #388 Centralized error handling, if appropriate?) That way, we will be able to reference them for future work.

Reply:
Hey @lauren-li, I have added an issue here: https://github.com/zowe/vscode-extension-for-zowe/issues/1346